src/tools/rustfmt/src/types.rs

```diff
@@ -463,8 +463,12 @@ impl Rewrite for ast::WherePredicate {
 
     fn rewrite_result(&self, context: &RewriteContext<'_>, shape: Shape) -> RewriteResult {
         // FIXME: dead spans?
-        let result = match *self {
-            ast::WherePredicate::BoundPredicate(ast::WhereBoundPredicate {
+        let mut result = self
+            .attrs
+            .rewrite_result(context, shape)
+            .unwrap_or(String::new());
```

Comment:
This doesn't seem right to me. If rustfmt fails to rewrite the attrs I'm worried that would result in the attrs being removed from the source code.

Also, can you add a formatting test case in `compiler/src/tools/rustfmt/tests`

Reply:
Do we need tests both in `rustfmt/tests/source/` and `rustfmt/tests/target`?

Reply:
As long as the formatting is idempotent you can just add a `.rs` file to `rustfmt/tests/target`